lcm-cmake/cpack.cmake

```diff
@@ -2,23 +2,43 @@
 set(PACKAGE_RELEASE_VERSION 1)
 
 # Default CPack generators
-set(CPACK_GENERATOR "TGZ;STGZ")
-
-# Determine distribution
-execute_process(COMMAND lsb_release -si
-	OUTPUT_VARIABLE LINUX_DISTRO
-	OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-# Determine architecture
-execute_process(COMMAND uname -m
-	OUTPUT_VARIABLE	MACHINE_ARCH
-	OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-# Add CPack generator according to detected distribution
-if(LINUX_DISTRO STREQUAL "Debian" OR LINUX_DISTRO STREQUAL "Ubuntu" OR LINUX_DISTRO STREQUAL "LinuxMint")
-    set(CPACK_GENERATOR "${CPACK_GENERATOR};DEB")
+set(CPACK_GENERATOR TGZ STGZ)
+
+# Detect OS type, OS variant and target architecture
+if(UNIX)
+    set(OS_TYPE Linux)
```

Comment:
Nit: should this also support MacOS? (`if(APPLE)`? Note that MacOS is a `UNIX` and does have `uname`, but no `lsb_release`.)